metrics/config_observability_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/system"
+
+	. "knative.dev/pkg/configmap/testing"
+	_ "knative.dev/pkg/system/testing"
+)
+
+func TestOurObservability(t *testing.T) {
```

Comment:
Sorry, I didn't look into the test file in detail before.

This specific test is testing the serving config file, i.e. `knative.dev/serving/config/config-observability.yaml`. You can find that `knative.dev/serving/pkg/metrics/testdata/config-observability.yaml` is a link to `knative.dev/serving/config/config-observability.yaml`. It should stay in `serving` repository.

Could you keep it in serving and move other tests which are testing `NewObservabilityConfigFromConfigMap` functionality to here?

@mattmoor WDYT?

Reply:
hmmm, yeah makes sense to split them.